pandas/tests/tslibs/test_array_to_datetime.py

```diff
@@ -206,38 +207,35 @@ def test_parsing_different_timezone_offsets():
 
 
 @pytest.mark.parametrize(
-    "invalid_date",
+    "invalid_date,exp_unit",
     [
-        date(1000, 1, 1),
-        datetime(1000, 1, 1),
-        "1000-01-01",
-        "Jan 1, 1000",
-        np.datetime64("1000-01-01"),
+        (date(1000, 1, 1), "s"),
+        (datetime(1000, 1, 1), "us"),
+        ("1000-01-01", "s"),
+        ("Jan 1, 1000", "s"),
+        (np.datetime64("1000-01-01"), "s"),
     ],
 )
 @pytest.mark.parametrize("errors", ["coerce", "raise"])
-def test_coerce_outside_ns_bounds(invalid_date, errors):
+def test_coerce_outside_ns_bounds(invalid_date, exp_unit, errors):
     arr = np.array([invalid_date], dtype="object")
-    kwargs = {"values": arr, "errors": errors}
 
-    if errors == "raise":
-        msg = "^Out of bounds nanosecond timestamp: .*, at position 0$"
+    result, _ = tslib.array_to_datetime(arr, errors=errors)
+    out_reso = np.datetime_data(result.dtype)[0]
+    assert out_reso == exp_unit
+    ts = Timestamp(invalid_date)
+    assert ts.unit == exp_unit
 
-        with pytest.raises(OutOfBoundsDatetime, match=msg):
-            tslib.array_to_datetime(**kwargs)
-    else:  # coerce.
-        result, _ = tslib.array_to_datetime(**kwargs)
-        expected = np.array([iNaT], dtype="M8[ns]")
-
-        tm.assert_numpy_array_equal(result, expected)
+    expected = np.array([ts._value], dtype=f"M8[{exp_unit}]")
+    tm.assert_numpy_array_equal(result, expected)
 
 
 def test_coerce_outside_ns_bounds_one_valid():
     arr = np.array(["1/1/1000", "1/1/2000"], dtype=object)
     result, _ = tslib.array_to_datetime(arr, errors="coerce")
 
-    expected = [iNaT, "2000-01-01T00:00:00.000000000"]
-    expected = np.array(expected, dtype="M8[ns]")
+    expected = ["1000-01-01T00:00:00.000000000", "2000-01-01T00:00:00.000000000"]
```

Comment:
Why did we have to remove `iNaT` here?

Reply:
in main the "1/1/2000" gets coerced to NaT, which is why iNaT is here.  With this PR, that coercion is no longer necessary